src/jit/optimizer.cpp

```diff
@@ -3332,7 +3332,7 @@ bool Compiler::optReachWithoutCall(BasicBlock* topBB, BasicBlock* botBB)
 
             // Does this block contain a gc safe point?
 
-            if (curBB->bbFlags & BBF_GC_SAFE_POINT)
+            if ((curBB->bbFlags & BBF_GC_SAFE_POINT) && !endsWithTailCall && (curBB != botBB))
```

Comment:
You change seem to be we can't trust the `BBF_GC_SAFE_POINT` flag when the block end with a tail call.
I don't like going down that route.

I think that we should explore the option of enforcing the invariant that having  `BBF_GC_SAFE_POINT` means that we have one or more normal non-tail callsites in the block.